### ISSUE TITLE
Data Format for sentence matching (How to Prepare the data?)

### ISSUE THREAD

**Issue by hana9090** [zihangdai/xlnet#60]
If we want to use xlnet for the text classification task, where the data format is text sentence and its label, how we can do that? I see that imdb task is classification but with different data format.

How to Prepare the data?

**Comment by kimiyoung** [comment]
You might use the IMDB processor but with your own data?

**Comment by jinamshah** [comment]
How does one replicate something like this from BERT to XLNet?
train_InputExamples = train.apply(lambda x: bert.run_classifier.InputExample(guid=None,
                                                                             text_a = x[DATA_COLUMN], 
                                                                             text_b = None, 
                                                                             label = x[LABEL_COLUMN]), axis = 1)

**Comment by kimiyoung** [comment]
Very similar. Please refer to `run_classifier.py` for details.

**Comment by jinamshah** [comment]
Okay thanks!
@kimiyoung  can you help me with this?
`model_fn = model_fn_builder(
    bert_config=modeling.BertConfig.from_json_file(CONFIG_FILE),
    num_labels=len(label_list),
    init_checkpoint=INIT_CHECKPOINT,
    learning_rate=LEARNING_RATE,
    num_train_steps=num_train_steps,
    num_warmup_steps=num_warmup_steps,
    use_tpu=True,
    use_one_hot_embeddings=True,
)`

I need to replicate this for XLNet.

**Comment by kimiyoung** [comment]
XLNet also has a similar function `get_model_fn` (see `run_classifier.py`).

**Comment by hana9090** [comment]
@kimiyoung what is the IMDB processor? if I need to apply xnel classification on quora questions how to prepare the data?

**Comment by kimiyoung** [comment]
Please refer to `run_classifier.py`. For other datasets, simply add one more data processor to it, which is similar to BERT.

**Comment by jinamshah** [comment]
> XLNet also has a similar function `get_model_fn` (see `run_classifier.py`).

The problem with that is when I start training, the XLRunConfig object needs a config file defined, which it tries to access from the flags. But the flags won't have any value as I am writing a python code and not a using command line
